crates/containerd-shim-wasm/src/sandbox/sync.rs

```diff
@@ -0,0 +1,117 @@
+use std::cell::OnceCell;
+use std::sync::{Arc, Condvar, Mutex};
+use std::time::Duration;
+
+/// A cell where we can wait (with timeout) for
+/// a value to be set
+pub struct WaitableCell<T> {
```

Comment:
I'd love to see some tests to ensure `WaitableCell<T>` behaves as expected in concurrent senarios.

Reply:
Will do!

Reply:
Added unit tests